pkg/acs/calacs/acs2d/dodark.c

```diff
@@ -97,19 +116,59 @@ int doDark (ACSInfo *acs2d, SingleGroup *x, float *meandark) {
   scicols = x->sci.data.nx;
   scirows = x->sci.data.ny; 
 
-  /* Compute DARKTIME */
-  /* SBC does not have FLASH keywords */
-  if (acs2d->detector == MAMA_DETECTOR)
-     darktime = acs2d->exptime;
-  else {
-     darktime = acs2d->exptime + acs2d->flashdur;
-
-     /* Post-SM4 non-BIAS WFC only */
-     /* TARGNAME unavailable, assume EXPTIME=0 means BIAS */
-     if (acs2d->detector == WFC_CCD_DETECTOR && acs2d->expstart > SM4MJD && acs2d->exptime > 0)
-        darktime += darkscaling;
+  /* Get the DARKTIME FITS keyword value stored in the ACSInfo structure */
+  darktimeFromHeader = (float)acs2d->darktime;
+
+  /*
+     The overhead offset time is a function of full-frame vs subarray and post-flash vs unflashed 
+     and has been determined empirically for CCD data.  Both the unflashed and post-flash
+     overhead values for full-frame or subarray were extracted from the calibration file 
+     during the table read.  Now it is just necessary to determine which offset actually
+     applies.
+  */
+
+  /* Unflashed observation */
+  darktimeOffset = acs2d->overhead_unflashed;
+ 
+  /* Post-flashed observation */
+  if ((acs2d->flashdur > 0.0) && (strcmp(acs2d->flashstatus, "SUCCESSFUL") == 0)) {
+        darktimeOffset = acs2d->overhead_postflashed;
   }
-  
+
+  /* 
+     Compute the final darktime based upon the date of full-frame or subarray data.
+     The full-frame overhead offset is applicable to all data post-SM4.  The subarray 
+     overhead offset is applicable to all data post-CYCLE24 and ONLY for supported
+     subarrays. 
+
+     Effectively the additive factor only applies to ACS/WFC as the HRC was no longer
+     operational by SM4MJD or CYCLE24, and SBC is a MAMA detector.
+  */
+  darktime = darktimeFromHeader;  /* Default */
+  if (acs2d->detector != MAMA_DETECTOR) {
+
+     /* Full-frame data */
+     if (acs2d->subarray == NO) {
+         if (acs2d->expstart >= SM4MJD)
```

Comment:
Standard best practice for if statements in C is to use the {} around the conditional code.